src/model.ts

```diff
@@ -236,6 +242,41 @@ export class GitExtension implements IGitExtension {
     return this._selectedHistoryFileChanged;
   }
 
+  /**
+   *  A signal emitted when the Git stash changes.
+   * 
+   */
+  get stashChanged(): ISignal <IGitExtension, IChangedArgs <Git.IStash>> {
+    return this._stashChanged;
+  }
+
+  /**
+   * Return the stash
+   */
+  get stash (): Git.IStash  {
+    return this._stash;
+  }
+
+  set stash(v: Git.IStash) {
```

Comment:
What is the idea behind this setter? I think we should not define a setter a the only way to update the stash should be through `refreshStash`.

```suggestion
  set stash(v: Git.IStash) {
```

Reply:
I use this in refreshStash (line 1597), so that I can set the stash. My understanding is that once I do `this.stash = stash`, it will call this `set stash` and then change the value and emit a signal if the stash value is different from the old value. Does my implementation make sense?